source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst

```diff
@@ -527,9 +527,9 @@ a `Connection`_ becomes available, re-entering the checkOut loop once it
 finishes waiting. This waiting MUST NOT block other threads from checking in a
 `Connection`_ to the pool. Threads that are waiting MUST be notified in order
 that they entered the WaitQueue. For drivers that implement the WaitQueue via a
-fair semaphore, a second semaphore may be required to implement this. Waiting on
-this second semaphore SHOULD be limited by the WaitQueueTimeout, if the driver
-supports one.
+fair semaphore, a condition variable may also be required to implement
+this. Waiting on the condition variable SHOULD be limited by the
+WaitQueueTimeout, if the driver supports one.
```

Comment:
Can you add test coverage for this timeout behavior? It should be possible by blocking two connection establishments longer than the configured waitQueueTimeoutMS (using failCommand block) and then running a 3rd checkout attempt which will timeout waiting on maxConnecting.